tests/ga/test_policy_engine.py

```diff
@@ -15,20 +15,406 @@
+
+    def test_policy_enforcement_should_be_disabled_when_conf_flag_false_or_no_policy_file(self):
+
+        # Test when conf flag is turned off - feature should be disabled.
+        self.patch_conf_flag.stop()
+        engine1 = ExtensionPolicyEngine()
+        self.assertFalse(engine1.is_policy_enforcement_enabled(),
+                         msg="Conf flag is set to false and policy file missing so policy enforcement should be disabled.")
+
+        # Turn on conf flag - feature should still be disabled, because policy file is not present.
+        self.patch_conf_flag.start()
+        engine2 = ExtensionPolicyEngine()
+        self.assertFalse(engine2.is_policy_enforcement_enabled(),
+                         msg="Policy file is not present so policy enforcement should be disabled.")
+
+        # Create a policy file, but turn off conf flag - feature should be disabled due to flag.
+        self.patch_conf_flag.stop()
+        self._create_policy_file({})
+        engine3 = ExtensionPolicyEngine()
+        self.assertFalse(engine3.is_policy_enforcement_enabled(),
                          msg="Conf flag is set to false so policy enforcement should be disabled.")
 
+    def test_should_raise_error_if_allowListedExtensionsOnly_is_string(self):
+        policy = \
+            {
+                "policyVersion": "0.1.0",
+                "extensionPolicies": {
+                    "allowListedExtensionsOnly": "True",  # Should be bool
+                    "signatureRequired": False,
+                    "extensions": {}
+                }
+            }
+        self._create_policy_file(policy)
+        with self.assertRaises(PolicyInvalidError, msg="String used instead of boolean, should raise error."):
+            ExtensionPolicyEngine()
+
+    def test_should_raise_error_if_signatureRequired_is_string(self):
+        policy_individual = \
+            {
+                "policyVersion": "0.1.0",
+                "extensionPolicies": {
+                    "allowListedExtensionsOnly": True,
+                    "signatureRequired": False,
+                    "extensions": {
+                        TEST_EXTENSION_NAME: {
+                            "signatureRequired": "False"  # Should be bool
+                        }
+                    }
+                }
+            }
+        policy_global = \
+            {
+                "policyVersion": "0.1.0",
+                "extensionPolicies": {
+                    "allowListedExtensionsOnly": True,
+                    "signatureRequired": "False",  # Should be bool
+                    "extensions": {}
+                }
+            }
+        for policy in [policy_individual, policy_global]:
+            self._create_policy_file(policy)
+            with self.assertRaises(PolicyInvalidError, msg="String used instead of boolean, should raise error."):
+                ExtensionPolicyEngine()
+
+    def test_extension_name_in_policy_should_be_case_insensitive(self):
```

Comment:
I think this unit test should be moved to TestExtensionPolicyEngine

Reply:
Done